### ISSUE TITLE
local'izing @DB::args leads to coredump

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11455]
Migrated from [rt.perl.org#93320](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=93320) (status was 'resolved')

Searchable as RT93320$

**Comment by p5pRT** [comment]
### From @dur-randir

#### Created by @dur-randir



The following script produces core dump on perls from 5\.8\.8 to 5\.14\.0

\#\!/usr/bin/perl
sub cl\{
&nbsp;    package DB;
&nbsp;    @&#8203;DB&#8203;::args = \(\);
&nbsp;    return caller\(shift\);
\}

sub f\{
&nbsp;    local @&#8203;DB&#8203;::args;
&nbsp;    my @&#8203;z = cl\($\_\) for \(1\.\.3\);
\}

f\(1\,2\,3\); f\(1\,2\,3\);

GDB stack trace is

\(gdb\) bt
\#0  0x0000000800b7e376 in memcpy \(\) from /lib/libc\.so\.7
\#1  0x00000000004bf4a1 in Perl\_pp\_caller \(\) at pp\_ctl\.c&#8203;:1965
\#2  0x000000000048ce23 in Perl\_runops\_standard \(\) at run\.c&#8203;:41
\#3  0x0000000000430417 in perl\_run \(my\_perl=Variable "my\_perl" is not
\#available\.
\) at perl\.c&#8203;:2350
\#4  0x000000000041cc69 in main \(argc=2\, argv=0x7fffffffe998\,
\# env=0x7fffffffe9b0\) at perlmain\.c&#8203;:120
\(gdb\) f 1
\#1  0x00000000004bf4a1 in Perl\_pp\_caller \(\) at pp\_ctl\.c&#8203;:1965
1965            Copy\(AvALLOC\(ary\)\, AvARRAY\(PL\_dbargs\)\, AvFILLp\(ary\) \+ 1 \+ off\, SV\*\);
\(gdb\) p PL\_dbargs
$1 = \(AV \*\) 0x703ac0
\(gdb\) p AvARRAY\(PL\_dbargs\)
$2 = \(SV \*\*\) 0x1

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=high

Site configuration information for perl 5.14.0:

Configured by inferno at Mon May 16 12:53:09 MSD 2011.

Summary of my perl5 (revision 5 version 14 subversion 0) configuration:
   
  Platform:
    osname=freebsd, osvers=7.3-stable, archname=amd64-freebsd
    uname='freebsd sigeon.ru 7.3-stable freebsd 7.3-stable #2: wed oct 13 22:55:27 msd 2010 inferno@sigeon.ru:usrobjusrsrcsyssigeon amd64 '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include',
    optimize='-O -g -ggdb3',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.2.1 20070719  [FreeBSD]', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags ='-Wl,-E  -fstack-protector -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lgdbm -ldb -lm -lcrypt -lutil -lc
    perllibs=-lm -lcrypt -lutil -lc
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fPIC', lddlflags='-shared  -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.14.0:
    /home/perltest/5.14.0-single-mymalloc/lib/site_perl/5.14.0/amd64-freebsd
    /home/perltest/5.14.0-single-mymalloc/lib/site_perl/5.14.0
    /home/perltest/5.14.0-single-mymalloc/lib/5.14.0/amd64-freebsd
    /home/perltest/5.14.0-single-mymalloc/lib/5.14.0
    .


Environment for perl 5.14.0:
    HOME=/home/game
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/home/inferno/bin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Now fixed with commit e1a80902e\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
